### PR TITLE
added available soon at players catalog n empty state component

### PR DESCRIPTION
**Pull request by mati-1**
[PM20-212](https://playmakerpro.atlassian.net/browse/PM20-212)

- added available soon view with query param
- added empty state component

### CODE REVIEW

**Comment by mati-1** [comment]
im not sure bout empty state component, is a cva needed?

**Review by dejwi — CHANGES_REQUESTED** [review]
I'll try later to make it work with import and .src, or you can try if you want I think we may only need some ts type declaration

Ideally I think it should work like

```ts
import BinocularsAnimation from '@/animations/bionculars.lottie'

....

<EmptyStateMessage
    title="Już wkrótce dostępne"
    text="Lista jest w trakcie przygotowywania. Włącz powiadomienie, a my
    poinformujemy Cię, gdy będzie dostepna."
    lottieAnimation={BinocularsAnimation.src}
/>
```

**Comment by dejwi** [comment]
> im not sure bout empty state component, is a cva needed?

I feel like it's a overkill in this case

File: components/common/EmptyStateMessage/empty-state-message.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    {animationType === 'spinner' && (
+      <div className="mb-4">
+        <Spinner desktopSize="huge" size="extraLarge" />
+      </div>
+    )}
+
+    {icon &&
+      cloneElement(icon, {
+        className: 'w-[42px] h-[42px] sd:w-[56px] sd:h-[56px]',
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/common/EmptyStateMessage/empty-state-message.tsx:43]: i saw that when you did the w-px n you automatically converted to rem? lol

File: components/common/EmptyStateMessage/empty-state-message.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  title,
+  animationType,
+  icon,
+}: IProps) => (
+  <div className="flex flex-col items-center justify-center gap-5">
+    {animationType === 'binoculars' && (
+      <div className="w-[7.5rem]">
+        <DotLottiePlayer src="animations/binoculars.lottie" autoplay loop />
+      </div>
+    )}
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/common/EmptyStateMessage/empty-state-message.tsx:33]: nice :)

File: components/common/EmptyStateMessage/empty-state-message.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dejwi) [reply components/common/EmptyStateMessage/empty-state-message.tsx:43]: it was probably a name of a precreated class not a custom px value inserted
I created classes with name like 3px to just find them faster XD

File: components/common/EmptyStateMessage/empty-state-message.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    {animationType === 'spinner' && (
+      <div className="mb-4">
+        <Spinner desktopSize="huge" size="extraLarge" />
+      </div>
+    )}
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/common/EmptyStateMessage/empty-state-message.tsx:39]: good, i didnt think bout that

File: components/common/EmptyStateMessage/empty-state-message.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (mati-1) [reply components/common/EmptyStateMessage/empty-state-message.tsx:43]: xD 💀

File: components/common/EmptyStateMessage/empty-state-message.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+export const EmptyStateMessage = ({
+  buttonIntent = 'primary',
+  text,
+  title,
+  isLoading,
+  lottieAnimation,
+  icon,
+}: IProps) => (
+  <div className="flex flex-col items-center justify-center gap-5">
+    {lottieAnimation === 'binoculars' && (
```
Comment (dejwi) [inline components/common/EmptyStateMessage/empty-state-message.tsx:31]:
```suggestion
    {lottieAnimation && (
```

File: components/common/EmptyStateMessage/empty-state-message.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+          autoplay
+          loop
+        />
+      </div>
+    )}
+
+    {isLoading && (
+      <div className="mb-4">
+        <Spinner desktopSize="huge" size="extraLarge" />
+      </div>
```
Comment (dejwi) [inline components/common/EmptyStateMessage/empty-state-message.tsx:44]: loading needs to work as a single state of this component
image

**Review by dejwi — CHANGES_REQUESTED** [review]

File: components/common/EmptyStateMessage/empty-state-message.tsx (hunk 4 above)
Comment (mati-1) [reply components/common/EmptyStateMessage/empty-state-message.tsx:31]: why something like this? maybe we'll add more animation in this component in the future

File: components/common/EmptyStateMessage/empty-state-message.tsx (hunk 5 above)
Comment (dejwi) [reply components/common/EmptyStateMessage/empty-state-message.tsx:44]: don't forget loading title and maybe add optional loadingDescription ?

File: components/common/EmptyStateMessage/empty-state-message.tsx (hunk 4 above)
Comment (dejwi) [reply components/common/EmptyStateMessage/empty-state-message.tsx:31]: yea so why do we have to check this stricly, with more variants you would have to check every single one and it's pointless

Comment (mati-1) [reply components/common/EmptyStateMessage/empty-state-message.tsx:31]: fr, i forgot bout src XD

**Review by dejwi — APPROVED** [review]
picobello

**Pull request merged**